editor/src/clj/editor/gui.clj

```diff
@@ -810,8 +1091,123 @@
+            ;; This output is used in the getters for all layout-related
+            ;; properties. Since it only needs to consider the current layout,
+            ;; and will fall back on the raw property values from the default
+            ;; layout if there isn't a layout-specific override, we can make it
+            ;; faster than the full layout->prop->value output.
+            ;;
+            ;; All layout-property-setters explicitly invalidate this output, so
+            ;; it is safe to evaluate properties on ourselves and our override
+            ;; originals here.
+            (when (coll/not-empty current-layout)
+              (let [basis (:basis _evaluation-context)]
+                (loop [node-id _node-id
+                       layout-names layout-names
+                       prop->value (get layout->prop->override current-layout)]
+                  (if-let [original-node-id (g/override-original basis node-id)]
+                    (let [layout-names-for-original (g/node-value original-node-id :layout-names _evaluation-context)]
+                      (cond
+                        (g/error-value? layout-names-for-original)
+                        layout-names-for-original
+
+                        ;; If our scene introduces the current-layout, we don't
+                        ;; need to consider any more layout overrides from our
+                        ;; override originals. Instead, anything not overridden
+                        ;; for the current-layout by this point will use values
+                        ;; from our default layout, or values inherited from the
+                        ;; default layout in our override originals.
+                        (and (contains? layout-names current-layout)
+                             (not (contains? layout-names-for-original current-layout)))
+                        (merge (make-recursive-prop->value-for-default-layout basis node-id)
+                               prop->value)
+
+                        :else
+                        (let [layout->prop->override-for-original (g/node-value original-node-id :layout->prop->override _evaluation-context)]
+                          (if (g/error-value? layout->prop->override-for-original)
+                            layout->prop->override-for-original
+                            (recur original-node-id
+                                   layout-names-for-original
+                                   (merge (get layout->prop->override-for-original current-layout)
+                                          prop->value))))))
+
+                    ;; We've reached the override root. Anything not overridden
+                    ;; for the current-layout by this point will use values from
+                    ;; our default layout.
+                    (let [node (g/node-by-id basis node-id)]
+                      (merge (make-prop->value-for-default-layout node)
+                             prop->value))))))))
```

Comment:
The `layout->prop->value` and `prop->value` outputs do similar work, but `prop->value` can be more efficient by focusing on the `current-layout`. It's worth it, because it is used in the `value` `fnk` of every layout property. However, it could also be implemented as simply looking up the `current-layout` in `layout->prop->value`.